src/java.naming/share/classes/com/sun/naming/internal/ObjectFactoriesFilter.java

```diff
@@ -97,18 +136,58 @@ private ObjectFactoriesFilter() {
      }
 
     // System property name that contains the patterns to filter object factory names
-    private static final String FACTORIES_FILTER_PROPNAME = "jdk.jndi.object.factoriesFilter";
+    private static final String FACTORIES_FILTER_PROPNAME =
+            "jdk.jndi.object.factoriesFilter";
+
+    // System property name that contains the patterns to filter LDAP object factory
+    // names
+    private static final String LDAP_FACTORIES_FILTER_PROPNAME =
+            "jdk.jndi.ldap.object.factoriesFilter";
+
+    // System property name that contains the patterns to filter RMI object factory
+    // names
+    private static final String RMI_FACTORIES_FILTER_PROPNAME =
+            "jdk.jndi.rmi.object.factoriesFilter";
+
+    // Default system property value that allows the load of any object factory
+    // classes
+    private static final String DEFAULT_GLOBAL_SP_VALUE = "*";
+
+    // Default system property value that allows the load of any object factory
+    // class provided by the JDK LDAP provider implementation
+    private static final String DEFAULT_LDAP_SP_VALUE =
+            "java.naming/com.sun.jndi.ldap.**;!*";
+
+    // Default system property value that allows the load of any object factory
+    // class provided by the JDK RMI provider implementation
+    private static final String DEFAULT_RMI_SP_VALUE =
+            "jdk.naming.rmi/com.sun.jndi.rmi.**;!*";
+
+    // A system-wide global object factories filter constructed from the system
+    // property
+    private static final ObjectInputFilter GLOBAL_FILTER =
+            ObjectInputFilter.Config.createFilter(
+                    getFilterPropertyValue(FACTORIES_FILTER_PROPNAME,
+                                           DEFAULT_GLOBAL_SP_VALUE));
 
-    // Default system property value that allows the load of any object factory classes
-    private static final String DEFAULT_SP_VALUE = "*";
+    // A system-wide LDAP specific object factories filter constructed from the system
```

Comment:
Where does the IllegalArgumentException from ObjectInputFilter.create get handled or reported?
If the property value is illformed, the error should be enough to diagnose and correct the property.

Reply:
That is a good point - the current state of reporting for a malformed filter pattern can be improved:
First filter check with `check*Filter` throws `java.lang.ExceptionInInitializerError` with a cause set to `java.lang.IllegalArgumentException` with filter pattern error. But subsequent checks throw `java.lang.NoClassDefFoundError: Could not initialize class com.sun.naming.internal.ObjectFactoriesFilter`.

To address that one interface and two new records have been added to represent a factory filter state:
 - `ConfiguredFilter` - interface for representing a filter created with `ObjectInputFilter.create` from a pattern.
 - `ValidFilter implements ConfiguredFilter` - stores `ObjectInputFilter` constructed from a valid filter pattern.
 - `InvalidFilter implements ConfiguredFilter` - stores `IllegalArgumentException` generated by parsing of an invalid filter pattern. The stored `IAE` is used to report malformed filter pattern each time specific filter is consulted.